### PR TITLE
Convert next batch of tests to BDD tests

### PR DESCRIPTION
**Pull request by lonvia**
This converts the tests for stage-2 processing in the flex output. The tests still rely on the common flex style. The only difference to the cpp tests is that the tests now create properly sorted input.

### CODE REVIEW

**Review by joto — CHANGES_REQUESTED** [review]
I believe the checks from test-output-flex-way-relation-del.cpp are missing.

File: tests/bdd/flex/way-add.feature [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        Then table osm2pgsql_test_t2 contains exactly
+            | way_id |
+            | 12     |
+        Then table osm2pgsql_test_tboth contains exactly
+            | way_id |
+            | 13     |
+            | 14     |
+
+
+    Scenario: add to t2
```
[the remark this answers is not in the document]
Comment (lonvia) [reply tests/bdd/flex/way-add.feature]: Fixed.

File: tests/bdd/flex/way-change.feature [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            | way_id |
+            | 13     |
+            | 14     |
+
+        Given the OSM data
+            """
+            <input>
+            """
+        When running osm2pgsql flex with parameters
+            | --slim | -a |
```
[the remark this answers is not in the document]
Comment (lonvia) [reply tests/bdd/flex/way-change.feature:39]: They are all horizontal now.

**Comment by lonvia** [comment]
Added the missing file and merged the style fixes into the existing commits to make them more pretty.

**Review by joto — APPROVED** [review]

**Pull request merged**